flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -660,7 +570,25 @@ public int run(
 					"yarn application -kill " + applicationId.getOpt());
 				yarnCluster.disconnect();
 			} else {
-				runInteractiveCli(yarnCluster, true);
+				ScheduledThreadPoolExecutor scheduledExecutorService = new ScheduledThreadPoolExecutor(1);
```

Comment:
Yes it could be. That way, however, we support that we can use an arbitrary executor which is available (as you've mentioned for tests). Since refactoring wouldn't add much value, I'll keep it like this.

Reply:
ok